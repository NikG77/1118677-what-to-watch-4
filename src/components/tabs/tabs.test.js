import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs.jsx";
// import TabName from "../../const.js";

const film = {
  title: `One Flew Over the Cuckoo's Nest`,
  src: ``,
  poster: ``,
  ratingScore: 8,
  ratingCount: 200,
  director: ``,
  starring: [``, ``],
  genre: `Drama`,
  releaseDate: 2020,
  pictureBackground: ``,
  previewVideo: ``,
  duration: 100,
  backgroundColor: ``,
  videoLink: ``,
  description: ` `,
  isFavorite: false,
  id: 1,
};

it(`Should Tabs render correctly`, () => {
  const tree = renderer
    .create(<Tabs
      film={film}
      onItemClick={() => {}}
      activeItem={``}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
