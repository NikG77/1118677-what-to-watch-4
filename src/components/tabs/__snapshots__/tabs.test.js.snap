// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Quentin Tarantino
            </cite>
            <time
              className="review__date"
              dateTime="2014-10-05"
            >
              October 5, 2014
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Akira Kurosawa
            </cite>
            <time
              className="review__date"
              dateTime="2014-10-20"
            >
              October 20, 2014
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Clint Eastwood
            </cite>
            <time
              className="review__date"
              dateTime="2019-09-05"
            >
              September 5, 2019
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          6.7
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Alejandro Gonsales Inarritu
            </cite>
            <time
              className="review__date"
              dateTime="2014-11-05"
            >
              November 5, 2014
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              James Cameron
            </cite>
            <time
              className="review__date"
              dateTime="2018-01-05"
            >
              January 5, 2018
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Tom Ford
            </cite>
            <time
              className="review__date"
              dateTime="2020-01-01"
            >
              January 1, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          1.9
        </div>
      </div>
    </div>
  </div>,
]
`;
